### ISSUE TITLE
Add support ALT Linux platforms detect and network configuration

### ISSUE THREAD

**Issue by mastersin** [mitchellh/vagrant#8745]
ALT Linux based distributions use /etc/net network configuration system not supported currently by vagrant. There are several vagrant projects available for testing:
- https://github.com/mastersin/samba-stand

For latest Vagrant versions (later than 1.9.4) current BaseALT (http://ftp.altlinux.org/pub/distributions/ALTLinux/p8/images/) images not works:
```bash
[sin@paas samba-stand]$ vagrant -v
Vagrant 1.9.6
[sin@paas samba-stand]$ vagrant destroy -f && vagrant up
==> client.domain.alt: VM not created. Moving on...
==> server.domain.alt: Forcing shutdown of VM...
==> server.domain.alt: Destroying VM and associated drives...
Bringing machine 'server.domain.alt' up with 'virtualbox' provider...
Bringing machine 'client.domain.alt' up with 'virtualbox' provider...
==> server.domain.alt: Importing base box 'mastersin/basealt-p8-server-systemd'...
==> server.domain.alt: Matching MAC address for NAT networking...
==> server.domain.alt: Checking if box 'mastersin/basealt-p8-server-systemd' is up to date...
==> server.domain.alt: Setting the name of the VM: server.domain.alt
==> server.domain.alt: Clearing any previously set network interfaces...
==> server.domain.alt: Specific bridge 'eth0' not found. You may be asked to specify
==> server.domain.alt: which network to bridge to.
==> server.domain.alt: Available bridged network interfaces:
1) br0
2) virbr0
3) enp4s0
4) docker0
5) br-e0e40b769a1a
==> server.domain.alt: When choosing an interface, it is usually the one that is
==> server.domain.alt: being used to connect to the internet.
    server.domain.alt: Which interface should the network bridge to? 1
    server.domain.alt: Which interface should the network bridge to? 1
==> server.domain.alt: Preparing network interfaces based on configuration...
    server.domain.alt: Adapter 1: nat
    server.domain.alt: Adapter 2: bridged
    server.domain.alt: Adapter 3: intnet
==> server.domain.alt: Forwarding ports...
    server.domain.alt: 22 (guest) => 2222 (host) (adapter 1)
==> server.domain.alt: Running 'pre-boot' VM customizations...
==> server.domain.alt: Booting VM...
==> server.domain.alt: Waiting for machine to boot. This may take a few minutes...
    server.domain.alt: SSH address: 127.0.0.1:2222
    server.domain.alt: SSH username: vagrant
    server.domain.alt: SSH auth method: private key
    server.domain.alt: 
    server.domain.alt: Vagrant insecure key detected. Vagrant will automatically replace
    server.domain.alt: this with a newly generated keypair for better security.
    server.domain.alt: 
    server.domain.alt: Inserting generated public key within guest...
    server.domain.alt: Removing insecure key from the guest if it's present...
    server.domain.alt: Key inserted! Disconnecting and reconnecting using new SSH key...
==> server.domain.alt: Machine booted and ready!
==> server.domain.alt: Checking for guest additions in VM...
==> server.domain.alt: Setting hostname...
==> server.domain.alt: Configuring and enabling network interfaces...
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

/sbin/ifdown 'eth1'
/sbin/ifdown 'eth2'
mv -f '/tmp/vagrant-network-entry-eth1-1499011032-0' '/etc/sysconfig/network-scripts/ifcfg-eth1'
mv -f '/tmp/vagrant-network-entry-eth2-1499011032-1' '/etc/sysconfig/network-scripts/ifcfg-eth2'
/sbin/ifup 'eth1'
/sbin/ifup 'eth2'

Stdout from the command:



Stderr from the command:

mv: cannot move '/tmp/vagrant-network-entry-eth1-1499011032-0' to '/etc/sysconfig/network-scripts/ifcfg-eth1': No such file or directory
mv: cannot move '/tmp/vagrant-network-entry-eth2-1499011032-1' to '/etc/sysconfig/network-scripts/ifcfg-eth2': No such file or directory
ERROR: /sbin/ifup: interface configuration directory '/etc/net/ifaces/eth1' not found
ERROR: /sbin/ifup: interface configuration directory '/etc/net/ifaces/eth2' not found
```
This problem reproduced on CentOS 7 and ALT Platform 8 hosts with minimal Base ALT Platform 8 image as guest operating system:
- https://atlas.hashicorp.com/mastersin/boxes/basealt-p8-server
- https://atlas.hashicorp.com/mastersin/boxes/basealt-p8-server-systemd

**Comment by mastersin** [comment]
What can I do for current pull request?
https://github.com/mitchellh/vagrant/pull/8746

I'm using it locally, but needs to make public my virtual stands on ALT platform.

**Comment by chrisroberts** [comment]
Resolved via #8746 

Cheers!
